packages/tenant-process/src/services/tenantService.ts

```diff
@@ -431,6 +439,120 @@ export function tenantServiceBuilder(
       return updatedTenant;
     },
 
+    async verifyVerifiedAttribute({
+      tenantId,
+      tenantAttributeSeed,
+      organizationId,
+      correlationId,
+    }: {
+      tenantId: TenantId;
+      tenantAttributeSeed: ApiVerifiedTenantAttributeSeed;
+      organizationId: TenantId;
+      correlationId: string;
+    }): Promise<Tenant> {
+      logger.info(
+        `Verifying attribute ${tenantAttributeSeed.id} to tenant ${tenantId}`
+      );
+
+      if (organizationId === tenantId) {
+        throw verifiedAttributeSelfVerification();
+      }
+
+      const allowedStatuses = [
+        agreementState.pending,
+        agreementState.active,
+        agreementState.suspended,
+      ];
+      await assertVerifiedAttributeOperationAllowed({
+        producerId: organizationId,
+        consumerId: tenantId,
+        attributeId: unsafeBrandId(tenantAttributeSeed.id),
+        agreementStates: allowedStatuses,
+        readModelService,
+        error: attributeVerificationNotAllowed(
+          tenantId,
+          unsafeBrandId(tenantAttributeSeed.id)
+        ),
+      });
+
+      const targetTenant = await retrieveTenant(tenantId, readModelService);
+
+      const attribute = await retrieveAttribute(
+        unsafeBrandId(tenantAttributeSeed.id),
+        readModelService,
+        attributeKind.verified
+      );
+
+      const verifiedTenantAttribute = targetTenant.data.attributes.find(
+        (attr): attr is VerifiedTenantAttribute =>
+          attr.type === tenantAttributeType.VERIFIED && attr.id === attribute.id
+      );
+
+      // eslint-disable-next-line functional/no-let
+      let updatedTenant: Tenant = {
+        ...targetTenant.data,
+        updatedAt: new Date(),
+      };
```

Comment:
We could improve the way this logic is laid out, I don't like the reassignment of the updatedTenant

Reply:
I'm open to discuss